src/components/clusterexplorer/node.ts

```diff
@@ -81,7 +85,29 @@ export type ClusterExplorerNode =
     ClusterExplorerHelmHistoryNode |
     ClusterExplorerCustomNode;
 
+export type ClusterExplorerNodeV2 =
+    ClusterExplorerMessageNode |
+    ClusterExplorerClusterNode |
+    ClusterExplorerContextNode |
+    ClusterExplorerResourceFolderNode |
+    ClusterExplorerGroupingFolderNode |
+    ClusterExplorerResourceNode |
+    ClusterExplorerConfigurationValueNode |
+    ClusterExplorerHelmReleaseNode |
+    ClusterExplorerHelmHistoryNode |
+    ClusterExplorerCustomNode;
+
 export class ClusterExplorerNodeImpl {
     readonly nodeCategory = KUBERNETES_EXPLORER_NODE_CATEGORY;
+    customizedTreeItem?: vscode.TreeItem;
```

Comment:
I think I can't do it if we stick to the current solution. As said above if you have a better solution i'm all ears.
If you read the comment above you should know why i check the `customizedTreeItem` inside the `getTreeItem()`. I need to create this variable in the same way as it exists inside `ClusterExplorerNodeBase` type otherwise i get a lot of warning by typescript.